features/api-layer/core/src/main/java/org/opennms/features/apilayer/scv/SecureCredentialsVaultImpl.java

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2022 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.apilayer.scv;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import org.opennms.integration.api.v1.scv.Credentials;
+import org.opennms.integration.api.v1.scv.SecureCredentialsVault;
+import org.opennms.integration.api.v1.scv.immutables.ImmutableCredentials;
+
+/** Exposes SecureCredentialsVault via Integration API */
+public class SecureCredentialsVaultImpl implements SecureCredentialsVault {
```

Comment:
We could implicitly add a prefix to the aliases so that plugins would be limited to seeing credentials that are for themselves or other plugins. A similar approach was taken here: https://github.com/OpenNMS/opennms/pull/4343/files#diff-ee0a089e00bf920ee637f5c8a6c22709a2f4f8f524a89cade1e5f2a3bcecf0deR39

Reply:
done. It will prevent the plugins from accessing OpenNMS internals but not from spying on each other.